apollo-router/src/plugins/telemetry/config_new/connector/events.rs

```diff
@@ -46,92 +58,67 @@ pub(crate) fn new_connector_events(
     let custom_events = config
         .custom
         .iter()
-        .filter_map(|(event_name, event_cfg)| match &event_cfg.level {
-            EventLevel::Off => None,
-            _ => Some(CustomEvent {
-                inner: Mutex::new(CustomEventInner {
-                    name: event_name.clone(),
-                    level: event_cfg.level,
-                    event_on: event_cfg.on,
-                    message: event_cfg.message.clone(),
-                    selectors: event_cfg.attributes.clone().into(),
-                    condition: event_cfg.condition.clone(),
-                    attributes: Vec::new(),
-                    _phantom: Default::default(),
-                }),
-            }),
-        })
+        .filter_map(|(name, config)| CustomEvent::from_config(name, config))
         .collect();
 
     ConnectorEvents {
-        request: config.attributes.request.clone().into(),
-        response: config.attributes.response.clone().into(),
-        error: config.attributes.error.clone().into(),
+        request: StandardEvent::from_config(&config.attributes.request),
+        response: StandardEvent::from_config(&config.attributes.response),
+        error: StandardEvent::from_config(&config.attributes.error),
         custom: custom_events,
     }
 }
 
-impl Instrumented
-    for CustomEvents<
-        ConnectorRequest,
-        ConnectorResponse,
-        (),
-        ConnectorAttributes,
-        ConnectorSelector,
-    >
-{
-    type Request = ConnectorRequest;
-    type Response = ConnectorResponse;
-    type EventResponse = ();
-
-    fn on_request(&self, request: &Self::Request) {
+impl CustomEvents<ConnectorRequest, ConnectorResponse, (), ConnectorAttributes, ConnectorSelector> {
+    pub(crate) fn on_request(&mut self, request: &ConnectorRequest) {
         // Any condition on the request is NOT evaluated here. It must be evaluated later when
         // getting the ConnectorEventRequest from the context. The request context is shared
         // between all connector requests, so any request could find this ConnectorEventRequest in
         // the context. Its presence on the context cannot be conditional on an individual request.
-        if self.request.level() != EventLevel::Off {
-            request
-                .context
-                .extensions()
-                .with_lock(|lock| lock.insert(ConnectorEventRequest(self.request.clone())));
+        if let Some(request_event) = self.request.take() {
+            request.context.extensions().with_lock(|lock| {
+                lock.insert(ConnectorEventRequest {
+                    level: request_event.level,
+                    condition: Arc::new(Mutex::new(request_event.condition)),
```

Comment:
As noted earlier I don't think we really need this wrapping, but I don't have time to check so I move it here just to be 100% sure I'm not introducing regression here.